src/network/networkd-util.c

```diff
@@ -99,3 +99,44 @@ int config_parse_address_family_boolean_with_kernel(
 
         return 0;
 }
+
+int ip_port_from_string(const char *s, uint16_t *port) {
+        int r;
+
+        assert(s);
+        assert(port);
+
+        r = safe_atou16(s, port);
+        if (r < 0 || *port == 0)
+                return -EINVAL;
```

Comment:
as a general rule: please don't clobber return values on failure, also see CODING_STYLE. Specifically, this means: invoke safe_atou16() with a temporary variable as second argument, and write it to *port only on success.